src/main/frontend/components/block.cljs

```diff
@@ -707,11 +707,8 @@
 (rum/defc block-reference < rum/reactive
   db-mixins/query
   [config id label]
-  (when (and
-         (not (string/blank? id))
-         (util/uuid-string? id))
-    (let [block-id (uuid id)
-          block (db/pull-block block-id)
+  (when-let [block-id (parse-uuid id)]
```

Comment:
I wouldn't recommend helper fns that do this. I think this leads to more complexity as bugs occur further away from the original error and usually lead to secondary errors. The complexity also shows up in helper fns as they become full of type checks e.g. `(when (string? ...)`. For the example here, if we know `id` is nil we should handle it here e.g. `(some-> id parse-uuid)` rather than pass it further down the call stack